### PR TITLE
Add changes to be ready for headless MC

### PR DESCRIPTION
**Pull request by Berg1989**
Also updated Day 1 handouts UNTIL Checkout (Headless API prep stuff) Checkout part isn't done yet: https://docs.google.com/forms/d/1NR_0NQ5am_UiUHfJTL1tv2HABfE77Nm_CGsRy0C34Yo/edit

### CODE REVIEW

File: src/Ucommerce.Masterclass.Umbraco/Controllers/ConfirmationController.cs [hunk 1]
```diff
@@ -25,20 +26,20 @@ public ActionResult Index()
             if (orderGuidParameterFromQueryString.IsNullOrWhiteSpace())
                 return View(new PurchaseOrderViewModel());
 
-            Ucommerce.EntitiesV2.PurchaseOrder basket = GetBasket(orderGuidParameterFromQueryString);
+            var basket = _transactionLibrary.GetPurchaseOrder(Guid.Parse(orderGuidParameterFromQueryString));
```
Comment (MadsMagnusDue) [inline src/Ucommerce.Masterclass.Umbraco/Controllers/ConfirmationController.cs]: Is there a headless approach for getting the order?

File: src/Ucommerce.Masterclass.Umbraco/Headless/Models/AuthenticationModel.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        [JsonProperty("expires_in")]
+        public int ExpiresIn { get; set; }
+
+        [JsonProperty("redirect_uri")]
+        public string RedirectUri { get; set; }
+
+        [JsonProperty("refresh_token")]
+        public string RefreshToken { get; set; }
+
+        public DateTime ExpiresAt { get; set; }
```
Comment (MadsMagnusDue) [inline src/Ucommerce.Masterclass.Umbraco/Headless/Models/AuthenticationModel.cs:21]: missing json prop

File: src/Ucommerce.Masterclass.Umbraco/Headless/TransactionClient.cs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    {
+
+        public async Task<PaymentMethodsOutput> GetPaymentMethods(string cultureCode, string countryId,
+            string priceGroupId, CancellationToken ct)
+        {
+            var client = await AuthorizeClient(ct);
+            var errorMessage = string.Empty;
+
+            using (var request = new HttpRequestMessage(new HttpMethod("GET"),
+                       $"/api/v1/payment-methods?cultureCode={cultureCode}&countryId={countryId}&priceGroupId={priceGroupId}"))
```
Comment (MadsMagnusDue) [inline src/Ucommerce.Masterclass.Umbraco/Headless/TransactionClient.cs]: remove the url value = empty string

File: src/Ucommerce.Masterclass.Umbraco/Headless/TransactionClient.cs [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+        }
+
+        public async Task UpdateOrderLineQuantity(string cultureCode, int quantity, string sku, string variantSku,
+            string priceGroupId, string productCatalogId, string basketId,
+            CancellationToken ct)
+        {
+            var client = await AuthorizeClient(ct);
+            var errorMessage = string.Empty;
+
+            using (var request = new HttpRequestMessage(new HttpMethod("POST"), $"/api/v1/baskets/{basketId}/lines?"))
```
Comment (MadsMagnusDue) [inline src/Ucommerce.Masterclass.Umbraco/Headless/TransactionClient.cs]: remove the url value = empty string + below parameters

File: src/Ucommerce.Masterclass.Umbraco/Headless/TransactionClient.cs [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+        public async Task EditBillingInformation(string basketId, string city, string firstName, string lastName,
+            string postalCode, string line1, string countryId, string emailAddress, string state,
+            string mobilePhoneNumber,
+            string attention, string company, CancellationToken ct)
+        {
+            var client = await AuthorizeClient(ct);
+            var errorMessage = string.Empty;
+
+            using (var request =
```
Comment (MadsMagnusDue) [inline src/Ucommerce.Masterclass.Umbraco/Headless/TransactionClient.cs:130]: remove the url value = empty string + below parameters

File: src/Ucommerce.Masterclass.Umbraco/Composers/UcommerceComposer.cs [hunk 6]
```diff
@@ -1,21 +1,28 @@
 using Ucommerce.Api;
+using MC_Headless.Headless;
+using MC_Headless.Resolvers;
+using MC_Headless.Resolvers.Impl;
 using Umbraco.Core.Composing;
+using Umbraco.Core;
 
-namespace Ucommerce.Masterclass.Umbraco.Composers
+namespace MC_Headless.Composers
```
Comment (MadsMagnusDue) [inline src/Ucommerce.Masterclass.Umbraco/Composers/UcommerceComposer.cs]: Why is the namespace updated?

File: src/Ucommerce.Masterclass.Umbraco/Api/MasterClassPaymentMethodController.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+        [HttpPost]
+        public IHttpActionResult Update()
+        {
+            var basketId = "";
+
+            if (string.IsNullOrWhiteSpace(basketId))
+                throw new MissingBasketIdException("Couldn't read basket id from cookies.");
+
+            var selectedPaymentMethodCookie = new HttpCookie("SelectedPaymentMethodId", Guid.NewGuid().ToString());
```
Comment (MadsMagnusDue) [inline src/Ucommerce.Masterclass.Umbraco/Api/MasterClassPaymentMethodController.cs]: Is the Guid.NewGuid() replaced in an exercise?

File: src/Ucommerce.Masterclass.Umbraco/Headless/ITransactionClient.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Ucommerce.Headless.Domain;
+
+namespace MC_Headless.Headless
```
Comment (MadsMagnusDue) [inline src/Ucommerce.Masterclass.Umbraco/Headless/ITransactionClient.cs]: The namespace looks wrong

File: src/Ucommerce.Masterclass.Umbraco/Api/MasterClassPaymentMethodController.cs (hunk 7 above)
Comment (Berg1989) [reply src/Ucommerce.Masterclass.Umbraco/Api/MasterClassPaymentMethodController.cs]: Yes, that's the idea :) - I will put it in a variable instead and make it an empty string

**Review by MadsMagnusDue — APPROVED** [review]
LGTM

**Pull request merged**